src/extensions/GetStackTrace.ts

```diff
@@ -0,0 +1,66 @@
+import {
+    BrsString,
+    RoArray,
+    Callable,
+    ValueKind,
+    StdlibArgument,
+    Int32,
+    BrsInvalid,
+    BrsType,
+    isBrsString,
+} from "../brsTypes";
+import type { Location } from "../lexer";
+import { Interpreter } from "../interpreter";
+
+const INTERNAL_REGEX_FILTER = /\(internal\)/;
+
+function stringifyLocation(location: Location) {
+    return `${location.file}:${location.start.line}:${location.start.column}`;
+}
+
+/**
+ * Returns a stack trace in the format:
+ * [
+ *   filename:line:column,
+ *   ...
+ * ]
+ */
+export const GetStackTrace = new Callable("GetStackTrace", {
+    signature: {
+        args: [
+            new StdlibArgument("numEntries", ValueKind.Int32, new Int32(10)),
+            new StdlibArgument("excludePatterns", ValueKind.Dynamic, BrsInvalid.Instance),
+        ],
+        returns: ValueKind.Object,
+    },
+    impl: (interpreter: Interpreter, numEntries: Int32, excludePatterns: BrsType) => {
+        let stack = interpreter.stack;
+
+        // Filter out any files that the consumer doesn't want to include
+        if (excludePatterns instanceof RoArray) {
+            excludePatterns.getValue().forEach((pattern) => {
+                if (isBrsString(pattern)) {
+                    let regexFilter = new RegExp(pattern.value);
```

Comment:
I think we should mention in the documentation that the Javascript regex engine is used for exclusions. [Brightscript uses PCRE ](https://developer.roku.com/en-ca/docs/references/brightscript/components/roregex.md)which works different in some expressions.

Reply:
Hmmm great point. Do you think it'd make more sense to pass in an array of `roRegex` objects instead of strings? Or keep it as-is, but update the docs?

Reply:
Given that our `roRegex` implementation isn't PCRE compatible either 😅 I'd  just update the docs and keep it as it is

Reply:
> Given that our `roRegex` implementation isn't PCRE compatible either 😅 I'd just update the docs and keep it as it is

Agreed!  As long as the docs mention that this is an array of strings and not an array of `roRegex` objects I'm onboard 😃